### ISSUE TITLE
JVM provider ignores raw array value matchers

### ISSUE THREAD

**Issue by artamonovkirill** [DiUS/pact-jvm#698]
Given a groovy DSL body like:
```
withBody {
    rawArrayEqTo eachLike(1, equalTo('1'))
    regexpRawArray eachLike(1, regexp(~/.+/, '1'))
}
```
Pact  JSON is generated as:
```
{
  "body": {
    "rawArray": ["1"],
    "rawArrayEqTo": ["1"],
    "regexpRawArray": ["1"]
  },
  "matchingRules": {
    "body": {
      "$.rawArrayEqTo": {
        "matchers": [{
           "match": "type"
          }],
          "combine": "AND"
      },
      "$.rawArrayEqTo[*]": {
        "matchers": [{
          "match": "equality"
        }],
        "combine": "AND"
      },
      "$.regexpRawArray": {
        "matchers": [{
          "match": "type"
        }],
        "combine": "AND"
      },
      "$.regexpRawArray[*]": {
        "matchers": [{
          "match": "regex",
          "regex": ".+"
        }],
        "combine": "AND"
      }
    }
  }
}
```
what is as expected.

But then on the provider side, a JSON like:
```
{
  "rawArrayEqTo": [""],
  "regexpRawArray": [""]
}
```
successfully passes Pact validation.

Consumer/provider test code: https://github.com/artamonovkirill/pact-matchers-issues/tree/master/src/test/groovy/com/github/artamonovkirill/pact/raw

**Comment by uglyog** [comment]
That is the correct behaviour. Use `minLike` if you want to fail validation on empty lists.

**Comment by uglyog** [comment]
With this fix, that JSON will now result in the following failure:

```
Failures:

0) Verifying a pact between arrays and ArraysProvider - get some arraysVerifying a pact between arrays and ArraysProvider - get some arrays returns a response which has a matching body
      $.rawArrayEqTo.0 -> Expected '' to equal '1'

      $.regexpRawArray.0 -> Expected '' to match '.+'
```

**Comment by uglyog** [comment]
Fix has been released with version 3.5.18
